packages/mobile/src/identity/contactMapping.ts

```diff
@@ -259,13 +257,24 @@ function* fetchWalletAddresses(e164Number: string) {
 
 // Returns a list of account addresses for the identifier received.
 export function* lookupAccountAddressesForIdentifier(id: string) {
+  const channel = yield call(lostAccountsChannel)
+  let lostAccounts: string[] = []
+  if (channel) {
+    const list = yield take(channel)
+    lostAccounts = list.map((address) => address.toLowerCase())
+  }
+
```

Comment:
Can you split out the logic regarding filtering of `lostAccounts` into its own generator function? I think this will make what is happening a little more clear